src/views/PurchaseOrderDetail.vue

```diff
@@ -139,12 +145,27 @@ export default defineComponent({
         }]
       });
       return alert.present();
+    },
+    updateProductCount(payload: any){
+      this.store.dispatch('purchaseOrder/updatePoProductCount', payload)
+    },
+    async scanCode(){
+      const modal = await modalController
+        .create({
+          component: Scanner,
+        });
+        modal.onDidDismiss()
+          .then((result) => {
+            this.updateProductCount(result.role);
```

Comment:
Why we are calling a method here and defining a new method? We can directly dispatch the action from here.